tests/unit/zhmcclient/test_notification.py

```diff
@@ -244,3 +262,131 @@ def test_one_message(self):
+        conn = receiver._conn  # pylint: disable=protected-access
+
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic1')
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic2')
+
+        result = receiver.is_subscribed('fake-topic1')
+        assert result is True
+
+        result = receiver.is_subscribed('foo')
+        assert result is False
+
+    @patch(target='stomp.Connection', new=MockedStompConnection)
+    def test_get_subscription(self):
+        """Test function for get_subscription() method."""
+
+        receiver = NotificationReceiver(self.topics, self.hmc, self.userid,
+                                        self.password)
+        conn = receiver._conn  # pylint: disable=protected-access
+
+        id_value1 = receiver.get_subscription('fake-topic1')
+        id_value2 = receiver.get_subscription('fake-topic2')
+        assert id_value1 != id_value2
+
+        # Check that the subscriptions are still in place
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic1')
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic2')
+
+    @patch(target='stomp.Connection', new=MockedStompConnection)
+    def test_get_subscription_nonexisting(self):
+        """Test function for get_subscription() method for a non-existing
+        subscription.."""
+
+        receiver = NotificationReceiver(self.topics, self.hmc, self.userid,
+                                        self.password)
+        conn = receiver._conn  # pylint: disable=protected-access
+
+        with pytest.raises(SubscriptionNotFound):
+            receiver.get_subscription('bla')
+
+        # Check that the subscriptions are still in place
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic1')
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic2')
+
+    @patch(target='stomp.Connection', new=MockedStompConnection)
+    def test_subscribe(self):
+        """Test function for subscribe() method."""
+
+        receiver = NotificationReceiver(self.topics, self.hmc, self.userid,
+                                        self.password)
+        conn = receiver._conn  # pylint: disable=protected-access
+
+        receiver.subscribe('foo')
+
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('foo')
+
+        # Check that the subscriptions are still in place
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic1')
+        # pylint: disable=no-member
+        assert conn.mock_get_subscription('fake-topic2')
+
+    @patch(target='stomp.Connection', new=MockedStompConnection)
+    def test_unsubscribe(self):
+        """Test function for unsubscribe() method."""
+
+        receiver = NotificationReceiver(self.topics, self.hmc, self.userid,
```

Comment:
receiver & conn are always (?) initialized the same way; couldn't that go into the setup function?

Reply:
DONE, also for the existing test classes.